test/sys/test_ptrace.rs

```diff
@@ -105,3 +105,78 @@ fn test_ptrace_cont() {
         },
     }
 }
+
+// ptrace::{setoptions, getreges} are only available in this platform
+#[cfg(all(target_os = "linux",
+          any(target_arch = "x86_64",
+              target_arch = "x86"),
+          target_env = "gnu"))]
+#[test]
+fn test_ptrace_syscall() {
+    use nix::sys::signal::kill;
+    use nix::sys::ptrace;
+    use nix::sys::signal::Signal;
+    use nix::sys::wait::{waitpid, WaitStatus};
+    use nix::unistd::fork;
+    use nix::unistd::getpid;
+    use nix::unistd::ForkResult::*;
+
+    let _m = ::FORK_MTX.lock().expect("Mutex got poisoned by another test");
+
+    // FIXME: qemu-user doesn't implement ptrace on all architectures
+    // and retunrs ENOSYS in this case.
```

Comment:
This is a little bit redundant.  On the one hand, you only compile the test on platforms that don't use qemu-user in CI.  OTOH, you try to detect qemu-user here.

Reply:
Good catch, thank you. I had copied this over from another test.